docs/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,7 @@
+# Why is this PR needed?
+
+# What does the PR do?
+
+# Does it break backwards compatibility?
+
+# List of future improvements not on this PR
```

Comment:
This heading is unclear to me. Is this referring to things we can still do relating to this PR to make it better?

Reply:
Yes, mainly refers to the technical debt you introduce that you wish you had the time to fix. 
Think of debt that are direcly related to your PR such as these:
- I introduced a `Tuple` library since I need to create a pair of 3 objects. We already have a `Pair` library and this is an overlap we should fix in the future
- This CSV file parser is needed only for migration but at the moment it can only parse file encoded as ASCII and not UTF-8. This should work for now but needs to be improved in the future

What do you think is a more appropriate heading for this? @fppt

Reply:
Maybe `Technical Debt To Be Addressed and Possible Optimisations To make`?

Reply:
sounds good to me. @aelred do you have any opinion on this matter?

Reply:
I'm pretty keen on keeping it short, so "Future Work" or "Future Improvements" is clear to me. But I'm OK with @fppt 's suggestion too.

Reply:
I am not super fussed. I just had to guess a little what `Future Work` meant. With no context it could mean:

1. Technical Debt Introduced in this PR
2. The next thing I am going to work on which may or may not be related to this PR
3. Performance improvements

Again, not super fussed.